app/components/quiz/QuizContext.js

```diff
@@ -0,0 +1,43 @@
+import React from 'react';
+import AppStore from 'store/appStore';
+
+export default React.createClass({
+  displayName: 'QuizContext',
+
+  contextTypes: {
+    router: React.PropTypes.func
+  },
+
+  propTypes: {
+    context: React.PropTypes.object.isRequired
+  },
+
+  render() {
+    var text = this.props.context.text;
+    var highlights = this.props.context.highlights;
+
+    var start = 0;
+    var tail = '';
+    var l = highlights.length;
+
+    if (highlights[l - 1][1] !== text.length) {
+      tail = <span>{text.substring(highlights[l - 1][1], text.length)}</span>;
+    }
+
+    return (
+      <p className="quiz__context">
+        {highlights.map((n, i) => {
+          // I'd rather not have to wrap these like this... is there a better way?
```

Comment:
oh, I'm referring to having
```
<span>
  <span />
  <span />
</span>
```
as opposed to being able to return two `<span>` as peers. it's a minor thing but the resulting markup on the page would be cleaner - I think there's a way to iterate so I can achieve this, but I couldn't figure out how without making my iteration significantly uglier.

anywho won't worry about it for now